### PR TITLE
Feature customserializers

### PR DESCRIPTION
**Pull request by thefex**
Implemented custom content deserialization - no breaking changes with exisiting api.
Issue #226

### CODE REVIEW

**Comment by ahmedalejo** [comment]
Any specific reason for re targeting?

**Comment by thefex** [comment]
project does not build/load on VS 2015 / Windows 10. Windows 8.0 platform isn't used anymore.

File: Refit/refit.nuspec [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
     <metadata>
         <id>refit</id>
-        <version>2.4.1</version>
+        <version>2.4.2</version>
```
Comment (ghuntley) [inline Refit/refit.nuspec]: This is a breaking change to Windows 8 Store and Windows 8 Phone users - bumping the patch is no the right digit to increment.

File: Refit/Refit-WinRT45.csproj [hunk 2]
```diff
@@ -12,12 +12,13 @@
     <RootNamespace>Refit</RootNamespace>
     <AssemblyName>Refit</AssemblyName>
     <DefaultLanguage>en-US</DefaultLanguage>
-    <TargetPlatformVersion>8.0</TargetPlatformVersion>
-    <MinimumVisualStudioVersion>11</MinimumVisualStudioVersion>
+    <TargetPlatformVersion>8.1</TargetPlatformVersion>
```
Comment (ghuntley) [inline Refit/Refit-WinRT45.csproj]: ReactiveUI as of v7 has dropped `WPA8` and `WP8` as the userbase is small and the projects do not open in Visual Studio. Essentially the path taken for that project is to keep it around (if painless) until Visual Studio (latest) no longer opens the project. So. I'm a-okay with dropping this platform and retargeting.

**Comment by onovotny** [comment]
Will close this for now as discussed in #226. I think the attribute-defined approach is cleaner and will gladly accept a PR for that.

**Pull request closed without merging**